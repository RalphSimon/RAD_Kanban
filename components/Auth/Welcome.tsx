export const Welcome = ({ children }) => (
  <h1 className="text-preset-2 welcome">
    {children}
    <style jsx>{`
			.welcome {
				text-align: center;
				color: var(--color-brand);
				margin-bottom: 24px;
			}
		`}</style>
  </h1>
)
