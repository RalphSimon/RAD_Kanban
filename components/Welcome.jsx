export const Welcome = () => (
	<div className="welcome">
		<div>
			<h1>Welcome</h1>
			<p>
				Delete the <pre>&#60;Welcome /&#62;</pre> component to start rolling your own
			</p>
		</div>
		<style jsx global>{/* CSS */ `
			*,
			*::before,
			*::after {
				box-sizing: border-box;
			}

			body {
				margin: 0;
				font-family: 'Avenir', 'Helvetica Neue', sans-serif;
				font-weight: 300;
				color: #212529;
			}

			.welcome {
				display: flex;
				align-items: center;
				justify-content: center;
				width: 100vw;
				height: 100vh;
				text-align: center;
			}

			h1 {
				font-size: 3rem;
			}

			p {
				font-size: 1rem;
			}
		`}</style>
	</div>
)
