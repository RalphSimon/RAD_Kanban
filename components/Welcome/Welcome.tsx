import * as React from 'react'

interface WelcomeProps {
  children: JSX.Element[] | JSX.Element;
}

export const Welcome = ({ children }: WelcomeProps) => (
  <div className="welcome">
    <div>
      <h1>Welcome</h1>
      <div>
				Delete the <pre>&#60;Welcome /&#62;</pre> component to start rolling
				your own
      </div>
    </div>

    <div className="welcome-switch">{children}</div>
    <style jsx global>{`
			*,
			*::before,
			*::after {
				box-sizing: border-box;
			}

			body {
				margin: 0;
				font-family: 'Avenir', 'Helvetica Neue', sans-serif;
				font-weight: 300;
				color: #212529;
			}

			.welcome,
			.welcome-switch {
				display: flex;
				flex-direction: column;
				align-items: center;
				justify-content: center;
				text-align: center;
			}

			.welcome {
				width: 100vw;
				height: 100vh;
			}

			.welcome-switch {
				width: 50%;
				border-top: 1px solid #212529;
				padding-top: 24px;
			}

			h1 {
				font-size: 3rem;
			}

			p {
				font-size: 1rem;
			}
		`}</style>
  </div>
)
