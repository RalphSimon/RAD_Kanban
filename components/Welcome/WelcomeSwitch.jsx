import { useState } from 'react'

export const WelcomeSwitch = () => {
	const [isChecked, setIsChecked] = useState(false)

	return (
		<label>
			<input
				type="checkbox"
				checked={isChecked}
				onChange={() => setIsChecked(state => !state)}
			/>
			<span>{isChecked ? 'on' : 'off'}</span>
		</label>
	)
}
