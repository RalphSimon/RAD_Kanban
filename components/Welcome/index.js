export { Welcome } from './Welcome'
export { WelcomeSwitch } from './WelcomeSwitch'
