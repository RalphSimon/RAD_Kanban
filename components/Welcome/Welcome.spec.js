/* eslint-disable no-undef */

import React from 'react'
import { Welcome } from './Welcome'
import renderer from 'react-test-renderer'

/*
	Straight from:
	https://jestjs.io/docs/en/tutorial-react
*/

it('gives an example of a Snapshot test', () => {
	const component = renderer.create(<Welcome />)

	let tree = component.toJSON()
	expect(tree).toMatchSnapshot()
})
