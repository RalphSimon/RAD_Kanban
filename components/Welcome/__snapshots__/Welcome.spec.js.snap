// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gives an example of a Snapshot test 1`] = `
<div
  className="jsx-4191556392 welcome"
>
  <div
    className="jsx-4191556392"
  >
    <h1
      className="jsx-4191556392"
    >
      Welcome
    </h1>
    <p
      className="jsx-4191556392"
    >
      Delete the 
      <pre
        className="jsx-4191556392"
      >
        &lt;Welcome /&gt;
      </pre>
       component to start rolling your own
    </p>
  </div>
  <div
    className="jsx-4191556392 welcome-switch"
  />
</div>
`;
